src/actions/RequestReviewActions.js

```diff
@@ -0,0 +1,66 @@
+// @flow
+
+import React from 'react'
+import { Linking, Platform } from 'react-native'
+import * as StoreReview from 'react-native-store-review'
+import { sprintf } from 'sprintf-js'
+
+import { TwoButtonSimpleConfirmationModal } from '../components/modals/TwoButtonSimpleConfirmationModal'
+import { Airship } from '../components/services/AirshipInstance'
+import s from '../locales/strings'
+import * as CORE_SELECTORS from '../modules/Core/selectors'
+import type { State } from '../types/reduxTypes.js'
+
+const SWAP_COUNT_DATA_FILE = 'swapCountData.json'
+const MANY_SWAPS_TO_TRIGGER_REQUEST = 3
+
+const requestReview = async () => {
+  if (Platform.OS === 'ios') {
+    StoreReview.requestReview()
+  } else if (Platform.OS === 'android') {
+    const title = sprintf(s.strings.request_review_question_title, s.strings.app_name_short)
+    await Airship.show(bridge => (
+      <TwoButtonSimpleConfirmationModal
+        bridge={bridge}
+        title={title}
+        subTitle={s.strings.request_review_question_subtitle}
+        cancelText={s.strings.request_review_answer_no}
+        doneText={s.strings.request_review_answer_yes}
+        onCancel={() => {}}
+        onDone={() => Linking.openURL(s.strings.request_review_android_page_link)}
+      />
+    ))
+  } else {
+    console.warn(`Unhandled Platform.OS: ${Platform.OS}. Unable to request review from user`)
+  }
+}
+
+export const updateSwapCount = async (state: State) => {
+  const account = CORE_SELECTORS.getAccount(state)
+  let swapCountData
+  try {
+    const swapCountDataStr = await account.disklet.getText(SWAP_COUNT_DATA_FILE)
+    swapCountData = JSON.parse(swapCountDataStr)
+  } catch (e) {
+    // File needs init
+    swapCountData = {
+      swap_count: 0,
+      has_review_been_requested: false
+    }
+  }
+
+  const hasReviewBeenRequested = swapCountData.has_review_been_requested
+  if (!hasReviewBeenRequested) {
+    let swapCount = parseInt(swapCountData.swap_count)
+    swapCount++
+    swapCountData.swap_count = swapCount
+    if (swapCount >= MANY_SWAPS_TO_TRIGGER_REQUEST) {
+      await requestReview()
+      swapCountData.has_review_been_requested = true
+    }
+    const swapCountDataStr = JSON.stringify(swapCountData)
+    await account.disklet
+      .setText(SWAP_COUNT_DATA_FILE, swapCountDataStr)
+      .catch(e => console.error(`RequestReviewActions.updateSwapCount: Error writing file ${SWAP_COUNT_DATA_FILE}:`, e))
```

Comment:
The only correct way to handle errors is `showError`, from `AirshipInstance.js`. Otherwise, if you don't want to bother the user with this, please put a comment saying that failure is OK, in which case it's `console.log`, not `console.error` for the logging.